loom/engine/bindings/loom/lmCSS.cpp

```diff
@@ -0,0 +1,1178 @@
+            _onUpdateDelegate.invoke();
+        }
+
+        /*
+         * Clears all the styles (rules). Attributes and the name are left unchanged,
+         * they are not dependant on the source of the stylesheet.
+         */
+        void clear()
+        {
+            utHashTableIterator<utHashTable<utHashedString, utArray<Style *> > > i(_styles.iterator());
+            while (i.hasMoreElements())
+            {
+                utArray<Style *>& styles = i.peekNextValue();
+
+                for (UTsize j = 0; j < styles.size(); j++)
+                {
+                    lmDelete(NULL, styles[j]);
+                }
+
+                i.next();
+            }
+
+            _styles.clear();
+        }
+
+        /*
+         * Checks the stylesheet for the presence of a style with a specific selector.
+         * Returns true if it was found, false otherwise.
+         */
+        bool hasStyle(const char* selector)
+        {
+            return _styles.find(utHashedString(selector)) != UT_NPOS;
+        }
+
+        /*
+         * Return a string representation of this object. For now this is it's name.
+         */
+        const char* toString()
+        {
+            return _name.c_str();
+        }
+
+        /*
+         * Add a new style with the given name (selector). If an existing style with
+         * the same selector already exists, it's added to an array.
+         * There are multiple styles possible with the same selector, they might
+         * have different attributes.
+         * Note that the style's own selector doesn't matter at lookup.
+         */
+        void newStyle(const char* selector, Style* style)
+        {
+            utHashedString key(selector);
+            if (_styles.find(key) != UT_NPOS)
+            {
+                utArray<Style *>* styles = _styles.get(key);
+                styles->push_back(style);
+            }
+            else
+            {
+                utArray<Style *> styles;
+                styles.push_back(style);
+                _styles.set(key, styles);
+            }
+        }
+
+        /*
+         * Get a composite style from the given styleNames (selectors, separated with a space).
+         * The styles found will be merged into a new style that will be returned.
+         */
+        int getStyle(lua_State* L)
+        {
+            const char* rawSelector = lua_tostring(L, -1);
+
+            // Get the LS type - we can't statically store it, that would fail with live assembly reload
+            Type* type = LSLuaState::getLuaState(L)->getType("loom.css.Style");
+
+            // Create an instance on top of the stack
+            lsr_createinstance(L, type);
+
+            // Copy the members
+            Style *result = (Style *)lualoom_getnativepointer(L, -1);
+            result->setName(rawSelector);
+
+            // Create a copy of the selector string we will be parsing
+            utString selector(rawSelector);
```

Comment:
It's wierd wrt memory managment. If we ever need it, we can come up with a solution for that case I think.